src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -404,6 +441,38 @@ private JsonRpcRequest ReadRequest(JToken json)
                 args is JArray argsArray ? (object)PartiallyParsePositionalArguments(argsArray) :
                 null;
 
+            // If method is $/progress, get the progress instance from the dictionary and call Report
+            string method = json.Value<string>("method");
+
+            if (string.Equals(method, ProgressRequestSpecialMethod, StringComparison.Ordinal))
+            {
+                try
+                {
+                    JToken progressId =
+                        args is JObject ? args["token"] :
+                        args is JArray ? args[0] :
+                        null;
+
+                    JToken value =
+                        args is JObject ? args["value"] :
+                        args is JArray ? args[1] :
+                        null;
+
+                    lock (this.progressLock)
+                    {
+                        if (this.progressMap.TryGetValue(progressId.Value<long>(), out ProgressParamInformation progressInfo))
```

Comment:
>progressId.Value<long>(), [](start = 57, length = 25)

We cannot assume the remote client is using numbers for their progress id. It can be any token. Can `progressMap` be changed to take the `TKey` for this dictionary as a generic type argument from this consuming class? Then this class can pass in `JToken` and the messagepack formatter (for which interop is not exactly defined, so we're free to define it) can use `long` directly.